### PR TITLE
Add missing news from #36146

### PR DESCRIPTION
**Pull request by oxinabox**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ New library functions
 New library features
 --------------------
 
-* `redirect_*`, functions now accept `DevNull` to discard all output redirected to it, and as an empty input. ([#36146]).
+* `redirect_*`, functions now accept `Base.DevNull` to discard all output redirected to it, and as an empty input. ([#36146]).
```
Comment (oxinabox) [inline NEWS.md:89]:
```suggestion
* `redirect_*`, functions now accept `devnull` to discard all output redirected to it, and as an empty input. ([#36146]).
```
`devnull` is exported though

**Comment by simeonschaub** [comment]
It's also missing a compat annotation, but that can also be a different PR.

**Comment by ViralBShah** [comment]
Do we want to get this in?

File: NEWS.md [hunk 2]
```diff
@@ -86,6 +86,7 @@ New library functions
 New library features
 --------------------
 
+* `redirect_*`, functions now accept `devnull` to discard all output redirected to it, and as an empty input. ([#36146]).
```
Comment (KristofferC) [inline NEWS.md:89]: To make it the same as the one below:

```suggestion
* The `redirect_*` functions now accept `devnull` to discard all output redirected to it, and as an empty input. ([#36146]).
```

but I don't understand "and as an empty input".

Comment (oxinabox) [reply NEWS.md]: It is empty when you read it.

```
julia> redirect_stdin(devnull) do
       println("*", readline(), "*")
       end
**
```

**Comment by ViralBShah** [comment]
Good to merge?

**Comment by StefanKarpinski** [comment]
Seems fine but has conflicts.

**Pull request merged**